tests/integration/conftest.py

```diff
@@ -159,6 +159,10 @@ def pytest_collection_modifyitems(
     This function implements a standard pytest hook. Please refer to pytest
     docs for further information.
     """
+    # do not try to skip tests if a keyword is provided (-k option)
+    if config.getoption("keyword"):
+        return
+
```

Comment:
Was this simple, huh? Nice! :)

That said, `keyword` is an internal implementation detail when it comes to option representation, only visible in a debugger, but not mentioned anywhere in the docs: https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.Config (you should link the docs in the commit message)

So you want to do `config.getoption("-k")` instead, proof:
```
ipdb> p config.getoption("-k")
'test_gomod.py or test_pip.py'
```

Since you're at it, you may as well expand the check to the `-m` option (as in `-m MARKEXPR`) as well. We're not currently using marks much (or at all), but it's another official way of filtering which we should consider.